sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java

```diff
@@ -302,7 +302,7 @@ private static NewIssueLocation newLocation(PythonInputFile inputFile, NewIssue
   }
 
   private void saveMeasures(PythonInputFile inputFile, PythonVisitorContext visitorContext) {
-    FileMetrics fileMetrics = new FileMetrics(visitorContext);
+    FileMetrics fileMetrics = new FileMetrics(visitorContext, inputFile instanceof GeneratedIPythonFile);
```

Comment:
Here you could use the `kind()` method on the inputFile instead of `instanceof`.